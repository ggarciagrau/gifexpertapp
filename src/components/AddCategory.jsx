import { useState } from "react";

export const AddCategory = ({ onNewCategory }) => {

    const [inputValue, setInputValue] = useState('');

    const onChange = ({ target }) => {
        setInputValue(target.value);
    }

    const onSubmit = (e) => {
        e.preventDefault();
        
        let cleanInputValue = inputValue.trim();
        if (!cleanInputValue.length) return;
        onNewCategory(cleanInputValue);
        setInputValue('');
        // setCategories(categories => [inputValue, ...categories]);
    }

    return (
        <div>
            <form onSubmit={onSubmit}>
                <input
                    type="text"
                    placeholder="Search gifs..."
                    value={inputValue}
                    onChange={onChange}
                />
            </form>
        </div>
    );
}