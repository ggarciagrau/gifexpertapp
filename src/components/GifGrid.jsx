import PropTypes from "prop-types";
import { GifGridItem } from "./GifGridItem";
import { useFetchGifs } from "../hooks/useFetchGifs";

export const GifGrid = ({ category }) => {

    const { images, isLoading } = useFetchGifs(category);

    return (
        <>
            <h3>{category}</h3>
            { isLoading && <h2>Cargando...</h2> }
            <div className="card-grid">
                {
                    images.map(({ id, title, url }) => (
                        <GifGridItem 
                            key={id}
                            {...{title, url, id}}
                        />
                    ))
                }
            </div>
        </>
    );
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}