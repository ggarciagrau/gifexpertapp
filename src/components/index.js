export * from './AddCategory';
export * from './GifGrid';
export *from './GifGridItem';

// import { AddCategory } from "./AddCategory";
// import { GifGrid } from "./GifGrid";
// import { GifGridItem } from "./GifGridItem";

// export {
//     AddCategory,
//     GifGrid,
//     GifGridItem
// };