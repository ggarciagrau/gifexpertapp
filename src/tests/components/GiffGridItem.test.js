import { render, screen } from "@testing-library/react";
import { GifGridItem } from "../../components";

describe("GifGridItem tests", () => {

    const title = "Google";
    const url = "https://google.es/";

    test("Renders", () => {

        const { container } = render(
            <GifGridItem
                title={title}
                url={url}
            />);

        expect(container).toMatchSnapshot();
    });

    test("Must display the image and the ALT attribute", () => {

        render(
            <GifGridItem
                title={title}
                url={url}
            />);
        // screen.debug();

        const { src, alt } = screen.getByRole('img');

        expect(src).toBe(url);
        expect(alt).toBe(title);
    });

    test("Must display the title", () => {

        render(
            <GifGridItem
                title={title}
                url={url}
            />);

        expect(screen.getByText(title)).toBeTruthy();
    });
});