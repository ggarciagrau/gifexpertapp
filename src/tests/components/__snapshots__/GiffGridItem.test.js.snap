// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGridItem tests Renders 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Google"
      src="https://google.es/"
    />
    <p>
      Google
    </p>
  </div>
</div>
`;
